File: src/main/java/application/history/history/application/services/DeleteHistoryService.java
package application.history.history.application.services;

import application.history.history.domain.repositories.HistoryRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDate;

@Service
public class DeleteHistoryService {
    private final HistoryRepository repository;

    public DeleteHistoryService(HistoryRepository repository) {
        this.repository = repository;
    }

    public void run(LocalDate date) {
        this.repository.deleteByDate(date);
    }
}

File: src/main/kotlin/com/example/history/application/services/DeleteHistoryService.kt
package com.example.history.application.services

import com.example.history.domain.repositories.HistoryRepository
import org.springframework.stereotype.Service
import java.time.LocalDate
import javax.transaction.Transactional

@Service
@Transactional
class DeleteHistoryService(repository: HistoryRepository) {
    private val repository: HistoryRepository

    fun run(date: LocalDate?) {
        repository.deleteByDate(date)
    }

    init {
        this.repository = repository
    }
}
